### PR TITLE
Concordia usernames

### PR DESCRIPTION
**Pull request by lassemoldrup**
## Purpose and Changes

- Uses the username from the database for Telegram checking.
- Bots now print the full name.
- Added some better error message for the frontend.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.

### CODE REVIEW

File: examples/bots/discord-bot/src/main.rs [hunk 1]
```diff
@@ -67,7 +67,7 @@ async fn help(
 }
 
 // Ephemeral: only the recipient can see the message
-/// Verify with Concordium
+/// Verify with Concordia
```
Comment (soerenbf) [inline examples/bots/discord-bot/src/main.rs:70]: Nice catch regarding Concordia vs Concordium! This is definitely more correct.

File: examples/bots/telegram-bot/src/main.rs [hunk 2]
```diff
@@ -159,14 +159,18 @@ async fn check_user(
...
+            let accounts = verification.accounts;
             let reply = if accounts.is_empty() {
-                format!("{name} is not verified with Concordium.")
+                format!("{name} is not verified with Concordia.")
             } else {
-                let mut reply = format!("{name} is verified with Concordium.");
+                let mut reply = format!("{name} is verified with Concordia.");
+                if let Some(full_name) = verification.full_name {
+                    reply.push_str("\n- Real name: ");
+                    reply.push_str(&format!("{} {}", full_name.first_name, full_name.last_name));
```
Comment (soerenbf) [inline examples/bots/telegram-bot/src/main.rs:172]: And here (if you decide to consolidate according to previous comment)

File: examples/some-verifier/frontend/src/components/Issuer.tsx [hunk 3]
```diff
@@ -30,12 +30,17 @@ interface DiscordWindowMessage {
...
+  telegramIssued,
+  setTelegramIssued,
+  discordIssued,
+  setDiscordIssued,
+}: {
+  telegramIssued: boolean;
+  setTelegramIssued: () => void;
+  discordIssued: boolean;
+  setDiscordIssued: () => void;
+}) {
```
Comment (soerenbf) [inline examples/some-verifier/frontend/src/components/Issuer.tsx:43]: Would prefer not to inline types 🙂 

```suggestion
interface IssuerProps {
  telegramIssued: boolean;
  setTelegramIssued: () => void;
  discordIssued: boolean;
  setDiscordIssued: () => void;
}

function Issuer({
  telegramIssued,
  setTelegramIssued,
  discordIssued,
  setDiscordIssued,
}: IssuerProps) {
```

File: examples/some-verifier/frontend/src/components/Issuer.tsx [hunk 4]
```diff
@@ -66,7 +71,7 @@ function Issuer() {
       const url = new URL(window.location.href);
       url.searchParams.set(Platform.Discord, 'true');
       window.history.replaceState(null, '', url);
```
Comment (soerenbf) [inline examples/some-verifier/frontend/src/components/Issuer.tsx:73]: I think we should move this (and the corresponding piece of code for telegram issuing) to the wrapping `App` component and the respective callbacks passed into the `Issuer` component. I think it's more clear what happens, if 1 component interfaces with URL query params 🙂

**Review by soerenbf — COMMENTED** [review]
Some minor stuff. Otherwise looks good 😄

File: examples/some-verifier/frontend/src/components/Issuer.tsx (hunk 4 above)
Comment (lassemoldrup) [reply examples/some-verifier/frontend/src/components/Issuer.tsx:73]: Good idea.

**Review by soerenbf — APPROVED** [review]

**Pull request merged**